src/epgsql_sock.erl

```diff
@@ -313,6 +327,16 @@ format_status(terminate, [_PDict, State]) ->
   State#state{rows = information_redacted}.
 
 %% -- internal functions --
+-spec send_socket_pasive(pg_sock()) -> pg_sock().
+send_socket_pasive(#state{subproto_state = #repl{receiver = Rec}} = State) when Rec =/= undefined ->
+    Rec ! {epgsql, self(), socket_passive},
+    State;
+send_socket_pasive(#state{subproto_state = #repl{ cbmodule = CbMod
+                                                , cbstate = CbState} = Repl
+                         } = State) ->
+    {ok, NewCbState} = epgsql:handle_socket_passive(CbMod, CbState),
```

Comment:
I think we should just do `setopts(State, [{active, ValueFromConnectOpts}])` here, because the pace would be synchronously regulated by the execution time of `handle_x_log_data/5` (the same way as it is going to work for any other `epgsql` / `epgsqla` but not `epgsqli` command).

This way we can be sure that the `epgsql`'s mailbox won't overflow if `handle_x_log_data/5` callback takes too long to complete

Reply:
If a socket is active, but the callback processes messages slow, it will result in a huge queue on `epgsql` side, right?

Reply:
By active you mean `{active, true}`? Yes. But `send_socket_passive` wont't be called at all when `{active, true}` because `ssl_passive`/`tcp_passive` won't be ever received.

Reply:
Let's for simplicity assume that one tcp/ssl message contains exactly one X-Log record. And user did `epgsql:connect(#{active => 3, replication => ".."})`.
And let's assume that postgres is able to produce hundreds of X-Log records per second, but `handle_x_log_data/5` can (synchronously) process 1 record per second.

It means epgsql would receive 3 X-Log packets from TCP and then immediately `tcp_passive` msg, so 4 messages in the queue.
It will call `handle_x_log_data/5` 3 times (once for each X-Log packet), it wil ltake 3 seconds.
And then it receives `tcp_passive` and calls `setopts(Sock, [{active, 3}])` and cycle repeats again.

There is never more than 4 messages in the mailbox at the same time.

(note that TCP/SSL messages not necessarily contain exactly one X-Log though)